### DISCUSSION TITLE
Show file and line number for every match item

### DISCUSSION THREAD

**Discussion by bigdogs** [BurntSushi/ripgrep#2220] [Q&A] [answered]
Hi, 
Thanks for the grate project. 

 Currently,  the `rg`'s default output is like this,
```
frontier\frontier_context.cpp
17:static constexpr int kFrontierServiceId = 20123;
24:static constexpr int kFrontierServiceId = 20183;
35:     service_id(kFrontierServiceId)
```

But I want to ask if the `rg` can display to the below format  (every item with a leading file and number). 
```
frontier\frontier_context.cpp:17 static constexpr int kFrontierServiceId = 20123;
frontier\frontier_context.cpp:24 static constexpr int kFrontierServiceId = 20183;
frontier\frontier_context.cpp:35     service_id(kFrontierServiceId)
```
The benefit is I can click the `file:line_number` to jump to file location from vs code console, that would be very useful for me.

**Comment by bigdogs** [comment]
emm...  `rg --no-heading` works. but there seems a little problem here.  the separator between `line number` and `matched text` is a colon.
e.g
```
file.cpp:23:void function();
```
which make vscode console  confusing about the file location.  as vscode think `void` is a column number

Not sure if there is a possible to replace the separator with other character (like a `tab` or `two spaces` or a line break)

**Comment by BurntSushi** [comment, marked as answer]
See #665 and #1994.

I'm willing to add something to ripgrep to solve this problem, but I can't add custom editor specific stuff for every editor. So I want to see one solution that works everywhere. Please contribute to the discussion in #665 if you can help with that.

Also, you do know that VS Code uses ripgrep for its "find in files" feature, right? I assume that has better integration. :)

**Comment by bigdogs** [comment]
Hi, Thanks for the reference.  It's a pity that the current implementation is so close to my needs. I  understand that it is  a huge work to support this feature for all console/editor. and thanks for your working on this. 

BTW. I think the  `rg` + `console` is more flexible and handy to use than the  "find in files", `rg` has more beautiful  key word highlight style, more context to display...
